src/js/d3_svg_cell_renderer.js

```diff
@@ -0,0 +1,207 @@
+				//	and <g>'s don't have dimension
+				var currVal = val;
+				if (typeof currVal === 'function') {
+					currVal = currVal(d,i);
+				}
+				if (typeof currVal === 'string' && currVal.indexOf('%') > -1) {
+					currVal = parseFloat(currVal)/100;
+					if (widthLike.indexOf(key) > -1) {
+						currVal = currVal*self.renderer.track.oncoprint.config.cell_width;	
+					} else if (heightLike.indexOf(key) > -1) {
+						currVal = currVal*self.renderer.config.cell_height;	
+					} 
+				}
+				return currVal+'';
+			});
+		});
+	};
+}
+
+function D3SVGCellRenderer(track) {
+	var self = this;
+	self.rule_set = new D3SVGRuleSet(self);
+	self.track = track;
+	self.config = self.track.config;
+	self.data = self.track.data;
+	self.cell_area;
+	self.svg;
+	self.g;
+
+	self.data_key = function(d) {
+		return d[self.config.id_member];
+	};
+
+	self.parseRuleSet = function(json_rules) {
+		self.rule_set.fromJSON(json_rules);
+	};
+
+	self.addRule = function(condition, d3_shape, attrs, z_index) {
+		var ret = self.rule_set.addRule(condition, d3_shape, attrs, z_index);
+		self.updateCells();
+		return ret;
+	};
+
+	self.removeRule = function(rule_id) {
+		self.rule_set.removeRule(rule_id);
+		self.updateCells();
+	};
+
+	self.renderCells = function(cell_area) {
+		self.cell_area = cell_area;
+
+		self.cell_area.selectAll('*').remove();
+		self.svg = self.cell_area.append('svg')
+		.attr('width', (self.track.oncoprint.config.cell_width + self.track.oncoprint.config.cell_padding)*self.track.data.length)
+		.attr('height', self.config.track_height);
+
+		self.g = self.svg.selectAll('g').data(self.data, self.data_key).enter().append('g').classed('cell', true);
+		self.updateCells();
+	};
+
+	self.updateCells = function() {
+		var config = self.config;
+		var id_order = utils.invert_array(self.track.oncoprint.id_order);
+		self.g.transition()
+		.attr('transform', function(d,i) {
+				return utils.translate(id_order[d[config.id_member]]*(self.track.oncoprint.config.cell_width + self.track.oncoprint.config.cell_padding), 0);
+			});
+
+		self.drawCells();
+	};
+
+	self.drawCells = function() {
+		self.g.selectAll('*').remove();
+
+		var renderRule = function(rule) {
+			rule.apply(self.g.data(rule.filterData(self.data), self.data_key));
+		};
+		var rule_lists = self.rule_set.getRules();
+		_.each(rule_lists, function(rule_list) {
+			_.each(rule_list, function(rule) {
+				renderRule(rule);
+			});
+		});
+		self.drawHitZones();
+	};
+
+	self.drawHitZones = function() {
+		var hits = self.g.append('rect').classed('hit', true)
+			.attr('width', self.config.cell_width)
+			.attr('height', self.config.cell_height)
+			.attr('stroke', 'rgba(0,0,0,0)')
+			.attr('fill', 'rgba(0,0,0,0)');
+		// bind events
+		var eventData = function(d, i, ctx) {
+			return {datum: d, index: i, track: self.track, g:d3.select(ctx.parentNode)};
+		}
+		hits.on('click', function(d, i){
+			$(self.track.oncoprint).trigger('cell_click.oncoprint', eventData(d,i,this));
```

Comment:
When writing new renderer implementations I think that it'd be nice to have a single place with essentially a list of all the events that are available to me to emit. Otherwise, I think that event names wind up being hard coded throughout the code base and harder to locate, hook into, etc.